src/network/gossip.rs

```diff
@@ -271,11 +282,13 @@ impl SnapchainGossip {
             system_tx,
             sync_channels: HashMap::new(),
             read_node,
-            bootstrap_addrs: config.bootstrap_addrs(),
+            bootstrap_addrs: config.bootstrap_addrs().into_iter().collect(),
             announce_address: config.announce_address.clone(),
             fc_network,
             contact_info_interval: config.contact_info_interval,
+            bootstrap_reconnect_interval: config.bootstrap_reconnect_interval,
             statsd_client,
+            connected_bootstrap_addrs: HashSet::new(),
```

Comment:
Do we need this? I think we can just call `self.swarm.connected_peers.contains`?

Reply:
We can't use that because we don't have peer id in the config. We need to track peer ids for each bootstrap address in order to use connected_peers which will require similar logic.

Reply:
The peer id is derived from the public key, which we know ahead of time for validators. https://github.com/libp2p/rust-libp2p/blob/master/identity/src/peer_id.rs#L126-L130

Anyway, not a big deal. This works as well